airflow/callbacks/callback_requests.py

```diff
@@ -55,7 +53,7 @@ class TaskCallbackRequest(CallbackRequest):
     def __init__(
         self,
         full_filepath: str,
-        simple_task_instance: SimpleTaskInstance,
+        simple_task_instance,  #: SimpleTaskInstance, Avoid circular import
```

Comment:
```suggestion
        simple_task_instance: "SimpleTaskInstance"
```

and then at the top of the file:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from airflow.models.taskinstance import SimpleTaskInstance
```

If you search for `if TYPE_CHECKING` you'll find more examples in the code base

Reply:
Thanks!